From: Takashi Iwai <tiwai@suse.com>
Subject: Disable nouveau for PCI device 10de:1cbb
References: bsc#1133593
Patch-mainline: never, temporary openSUSE-specific hack

nouveau driver causes a kernel panic on Dell Precision 5530 when the
runtime PM is enabled (as default).  Disable the probe for this particular
device on the installer kernel, at least.

Note that the bug was already fixed in the update kernel, so this patch
has to be removed again once after merging to the update git branch!

Signed-off-by: Takashi Iwai <tiwai@suse.com>
Signed-off-by: Petr Tesarik <ptesarik@suse.com>
---
 drivers/gpu/drm/nouveau/nouveau_drm.c |    7 +++++++
 1 file changed, 7 insertions(+)
--- a/drivers/gpu/drm/nouveau/nouveau_drm.c
+++ b/drivers/gpu/drm/nouveau/nouveau_drm.c
@@ -466,6 +466,13 @@ static int nouveau_drm_probe(struct pci_
 	bool boot = false;
 	int ret;
 
+	/* XXX disable nouveau for 10de:1cbb on Leap 15.1 GM (bsc#1133593) */
+	if (pdev->vendor == 0x10de && pdev->device == 0x1cbb) {
+		dev_err(&pdev->dev, "nouveau is DISABLED for this device!\n");
+		dev_err(&pdev->dev, "See https://bugzilla.opensuse.org/show_bug.cgi?id=1133593.\n");
+		return -ENODEV;
+	}
+
 	if (vga_switcheroo_client_probe_defer(pdev))
 		return -EPROBE_DEFER;
 
